### ISSUE TITLE
StreamPutData fails if no debugger is attached

### ISSUE THREAD

**Issue by luksal** [ManagedBass/ManagedBass#108]
I have a "push" stream set up like this:

```
_stream = Bass.CreateStream(48000, 2, BassFlags.Default, StreamProcedureType.Push);
```

 in my audio decoding function I'm using to push the PCM data to the stream:

```
Bass.StreamPutData(_stream, samples, samples.Length);
```
where samples is a byte array containing the decoded audio.

Unfortunately this works only, while the debugger is attached. Running the code without a debugging sessions results in a `NotAvailable` error.
Or when I try to play the empty stream using:
```
Bass.ChannelPlay(_stream)
```
a hard fault `Process finished with exit code 139.` occurs.

I'm pretty sure this has something to do with marshaling, but I don't know where I have to look to resolve this issue.

Btw.: stream creation from file works without a problem, even in release configuration.

**Comment by luksal** [comment]
The problem is that, `static extern int BASS_StreamCreate(int Frequency, int Channels, BassFlags Flags, StreamProcedureType ProcedureType, IntPtr User = default(IntPtr));` in `Streams.cs` line 80 uses a wrong signature.
The type of the Parameter `ProcedureType` has to be `IntPtr` to work properly.

Changing the code, to the following, works for me. I will prepare a pull request soon to apply the fix

```
static extern int BASS_StreamCreate(int Frequency, int Channels, BassFlags Flags, IntPtr ProcedureType, IntPtr User = default(IntPtr));
public static int CreateStream(int Frequency, int Channels, BassFlags Flags, StreamProcedureType ProcedureType)
{
    return BASS_StreamCreate(Frequency, Channels, Flags, new IntPtr((int)ProcedureType));
}
```

**Comment by olitee** [comment]
Closing this as it's been fixed with #109 

Thanks @luksal
